source/NVDAObjects/UIA/__init__.py

```diff
@@ -998,3 +1007,32 @@ def event_nameChange(self):
 	def event_stateChange(self):
 		return
 
+class SearchField(UIA):
+	"""An edit field that presents suggestions based on search term.
+	"""
+
+	def event_UIA_controllerFor(self):
+		# Only useful if suggestions appear and disappear.
+		focus = api.getFocusObject()
+		if len(self.controllerFor)>0:

```

Comment:
When I tested it with Settings and what not, just using self.controllerFor was enough, but I'd be happy to check the focus to really make sure (thanks for catching this).